ios/tunnel/tunnel.go

```diff
@@ -0,0 +1,328 @@
+package tunnel
```

Comment:
Please add a detailed description here about what this module is for.

Reply:
There's now a `doc.go` file for the package documentation with some detail. It's a first draft, and I'll probably update it again 😄

Reply:
ah, I think the golang convention is to put this here. I might be also wrong tho, let me know